File: src/main/java/me/nepnep/nepbot/message/command/commands/admin/SetDefaultRole.java
package me.nepnep.nepbot.message.command.commands.admin;

import me.nepnep.nepbot.Database;
import me.nepnep.nepbot.message.command.ICommand;
import me.nepnep.nepbot.message.command.Category;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.Role;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class SetDefaultRole implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {
        if (event.getMessage().getMentionedRoles().size() != 0) {
            Role role = event.getMessage().getMentionedRoles().get(0);

            Database.write(event.getGuild(), role.getName());
        } else {
            event.getChannel().sendMessage("Please mention a role").queue();
        }
    }
    @Override
    public String getInvoke() {
        return "setdefaultrole";
    }
    @Override
    public Permission requiredPermission() {
        return Permission.MANAGE_ROLES;
    }
    @Override
    public Category getCategory() {
        return Category.ADMIN;
    }
    @Override
    public String getDescription() {
        return "Sets the default role: ;setdefaultrole <Mention role>";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/admin/SetDefaultRole.kt
package me.nepnep.nepbot.message.command.commands.admin

import me.nepnep.nepbot.database.setDefaultRole
import me.nepnep.nepbot.message.command.Category
import me.nepnep.nepbot.message.command.AbstractCommand
import net.dv8tion.jda.api.Permission
import net.dv8tion.jda.api.entities.GuildMessageChannel
import net.dv8tion.jda.api.events.message.MessageReceivedEvent

class SetDefaultRole : AbstractCommand(
    "setdefaultrole",
    Category.ADMIN,
    "Sets the default role: ;setdefaultrole <Mention role> or empty arguments for unsetting",
    Permission.MANAGE_ROLES
) {
    override suspend fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        val mentionedRoles = event.message.mentions.roles
        val guild = event.guild
        if (mentionedRoles.isEmpty()) {
            guild.setDefaultRole(null)
        } else {
            guild.setDefaultRole(mentionedRoles[0])
        }
    }
}